tests/omp/test_omp_tridiag.f90

```diff
@@ -43,15 +51,30 @@ program test_omp_tridiag
 
   if (nrank == 0) print *, 'Parallel run with', nproc, 'ranks'
 
-  pnext = modulo(nrank - nproc + 1, nproc)
-  pprev = modulo(nrank - 1, nproc)
+  ! Global number of cells in each direction
+  dims_global = [1024, 64, 64]
 
-  n_glob = 1024
-  n = n_glob/nproc
-  n_block = 64*64/SZ
+  ! Global domain dimensions
+  L_global = [2*pi, 2*pi, 2*pi]
+
+  ! Domain decomposition in each direction
+  nproc_dir = [nproc, 1, 1]
+
+  mesh = mesh_t(dims_global, nproc_dir, L_global)
+  allocator = allocator_t(mesh, SZ)
+  
+  pnext = mesh%par%pnext(DIR_X)
+  pprev = mesh%par%pprev(DIR_X)
+
+  n_glob = dims_global(1)
+  n = mesh%get_n(DIR_X, VERT)
+  n_groups = mesh%get_n_groups(DIR_X)
   n_iters = 1
 
-  allocate (u(SZ, n, n_block), du(SZ, n, n_block))
+  u_field => allocator%get_block(DIR_X, VERT)
+  du_field => allocator%get_block(DIR_X, VERT)
+  u = u_field%data
+  du = du_field%data
```

Comment:
The idea here was that we sort out the `pprev`, `pnext` and allocation of the fields (arrays here actually) ourselves so that the tridiagonal solver can be tested in isolation in a unit test (including the sendrecv that tridiagonal solver require). Even if there is something wrong with the `allocator` or `mesh` we expect this test to run correctly as long as the low level tridiagonal solver algorithm is implemented correctly.